api/server/handler.go

```diff
@@ -121,14 +123,61 @@ func (m *HttpServer) handleBlobGetByID(w http.ResponseWriter, r *http.Request, i
 		return
 	}
 	defer blobReader.Close()
+
 	w.Header().Set(httpHeaderContentTypeOctetStream())
 	w.Header().Set(httpHeaderContentTypeOptionsNoSniff())
 	w.Header().Set(httpHeaderContentLength(blobDesc.Size))
 	w.Header().Set("Content-Disposition", fmt.Sprintf(`attachement; filename="%s.bin"`, id.String()))
-	http.ServeContent(w, r, "", blobDesc.ModificationTime, blobReader)
+
+	rangeHeader := r.Header.Get("Range")
+	if rangeHeader != "" {
+		// Get file size from reader.
+		size, err := blobReader.Seek(0, io.SeekEnd)
+		if err != nil {
+			respondWithJson(w, api.ErrorResponse{Error: err.Error()}, http.StatusBadRequest)
+			return
+		}
+		_, err = blobReader.Seek(0, io.SeekStart)
+		if err != nil {
+			respondWithJson(w, api.ErrorResponse{Error: err.Error()}, http.StatusBadRequest)
+			return
+		}
+
+		// If the reader has a WriteToN method, use it to do the copy.
+		// Avoids an allocation and a copy.
+		wtn, ok := blobReader.(WriterToN)
+		if ok {
+			ranges, err := httprange.ParseRange(rangeHeader, size)
+			if err != nil {
+				respondWithJson(w, api.ErrorResponse{Error: err.Error()}, http.StatusBadRequest)
+				return
+			}
+			for i := range ranges {
+				blobReader.Seek(ranges[i].Start, io.SeekStart)
+				_, err = wtn.WriteToN(w, ranges[i].Length)
```

Comment:
before we write, what here causes the server to respond with a `206` rather than a `200` status to indicate it's respecting the requested range(s)?

Reply:
NA, Sorry, this PR should have been draft.